addons/website_slides_survey/models/slide_slide.py

```diff
@@ -65,3 +65,28 @@ def _action_set_viewed(self, target_partner, quiz_attempts_inc=False):
                         'slide_partner_id': new_slide_partner.id
                     }
                 )
+
+    def _get_slide_certification_data(self):
+        certification_data = {}
+        if not all(slide.channel_id.is_member for slide in self):
+            for s in self:
+                user_input = s.survey_id._create_answer(
+                    partner=self.env.user.partner_id,
+                    check_attempts=False,
+                    test_entry=True, **{
+                        'slide_id': s.id
+                    }
+                )
+                certification_data[s.id] = user_input._get_survey_url()
+        else:
+            for slide in self:
+                if not self.env.user._is_public() and slide.slide_type == 'certification' and slide.survey_id:
```

Comment:
IMO you can change `not self.env.user._is_public()` with `slide.channel_id.is_member` and avoid too much nesting.

Reply:
Done